File: 백준/Silver/10815. 숫자 카드/숫자 카드.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        // 시간초과 해결해야 한다.
        // 예를 들어 contains와 같은 함수를 사용하는 것도 좋지만, binary search를 사용하는 것도 가능
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());
        int[] arr = new int[N];
        for(int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(arr);
        int M = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        for(int i = 0; i < M; i++) {
            int num = Integer.parseInt(st.nextToken());
            System.out.print(binarySearch(arr, num) + " ");
        }
    }
    public static int binarySearch(int[] arr, int key) {
        int low = 0;
        int high = arr.length - 1;
        while(low <= high) {
            int mid = (low + high) / 2;
            if(key > arr[mid]) {    // mid보다 클 때
                low = mid + 1;
            }else if(key < arr[mid]) {  // mid보다 작을 때
                high = mid - 1;
            }else{
                return 1;
            }
        }
        return 0;
    }
}

File: 백준/Silver/10815. 숫자 카드/숫자 카드.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);
	cout.tie(0);

    int N = 0;
    int M = 0;
    cin >> N;
    
    vector<int> v1(N);
    for(int i = 0; i < N; i ++){
        cin >> v1[i];
    }
    
    sort(v1.begin(), v1.end());

    cin >> M;
    for(int i = 0; i < M; i ++){
        int num = 0;
        cin >> num;
        if(binary_search(v1.begin(), v1.end(), num)){
            cout << "1" << " ";
        }else{
            cout << "0" << " ";
        }
    }


    return 0;
}